tg/util/sqlalchemy.py

```diff
@@ -3,11 +3,18 @@
 
 try:
     import sqlalchemy
-    from sqlalchemy.engine import ResultProxy, RowProxy
+    from sqlalchemy.engine import ResultProxy
 except ImportError:  # pragma: no cover
     sqlalchemy = None
     ResultProxy = None
-    RowProxy = None
+
+try:  # sa>=1.4
+    from sqlalchemy.engine import LegacyRow as RowProxy
+except ImportError:
+    try:  # sa<1.4
+        from sqlalchemy.engine import RowProxy
+    except ImportError:  # pragma: no cover
+        RowProxy = None
```

Comment:
Shouldn't we also change `is_query_row` to support `Row` too? Here it seems we only return it's a query row for the legacy rows, but not for the new ones. Which seems wrong.

Reply:
Maybe we should have a separate branch for TG2.5 where we start supporting sqalchemy with `future=True`?